app/bundles/LeadBundle/Entity/StatDevice.php

```diff
@@ -0,0 +1,365 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2014 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use JMS\Serializer\Annotation as Serializer;
+use Mautic\CoreBundle\Doctrine\Mapping\ClassMetadataBuilder;
+use Mautic\CoreBundle\Entity\IpAddress;
+use Mautic\CoreBundle\Helper\EmojiHelper;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\ApiBundle\Serializer\Driver\ApiMetadataDriver;
+/**
+ * Class StatDevice
+ *
+ * @package Mautic\LeadBundle\Entity
+ */
+class StatDevice
+{
+
+    /**
+     * @var int
+     */
+    private $id;
+
+    /**
+     * @var integer
+     */
+    private $stat;
+
+    /**
+     * @var string
+     */
+    private $channel;
+
+    /**
+     * @var integer
+     */
+    private $channelId;
+
+    /**
+     * @var \Mautic\CoreBundle\Entity\IpAddress
+     */
+    private $ipAddress;
+
+    /**
+     * @var \DateTime
+     */
+    private $dateOpened;
+
+    /**
+     * @var array
+     */
+    private $clientInfo = array();
+
+    /**
+     * @var string
+     */
+    private $device;
+
+    /**
+     * @var string
+     */
+    private $deviceOsName;
+
+    /**
+     * @var string
+     */
+    private $deviceOsShortName;
+
+    /**
+     * @var string
+     */
+    private $deviceOsVersion;
+
+    /**
+     * @var string
+     */
+    private $deviceOsPlatform;
+
+    /**
+     * @var string
+     */
+    private $deviceBrand;
+
+    /**
+     * @var string
+     */
+    private $deviceModel;
+
+
+    /**
+     * @param ORM\ClassMetadata $metadata
+     */
+    public static function loadMetadata (ORM\ClassMetadata $metadata)
+    {
+        $builder = new ClassMetadataBuilder($metadata);
+
+        $builder->setTable('lead_stats_devices')
+            ->setCustomRepositoryClass('Mautic\LeadBundle\Entity\StatDeviceRepository');
```

Comment:
Let's add some indexes on this table - based on what I'm seeing in the code, looks like we should have indexes date opened, stat_id, , channel + channel_id, and device.

Also, let's get a addLead() association here (and populate wherever this Entity is created) so we can search this table for a lead's devices without having to find all their stat hits first.